packages/plugins/experimental/plugin-map/src/hooks/useMarkers.ts

```diff
@@ -4,55 +4,32 @@
 
 import { useMemo } from 'react';
 
-import { getTypename, type DynamicSchema } from '@dxos/echo-schema';
-import { CollectionType } from '@dxos/plugin-space';
-import { TableType } from '@dxos/plugin-table/types';
+import { DynamicSchema, getTypename, StoredSchema } from '@dxos/echo-schema';
 import { Filter, getSpace, useQuery } from '@dxos/react-client/echo';
 
 import { type MapType, type MapMarker } from '../types';
 
 /**
- * Find table rows that are in the same collection as the map.
- * Return the ones with `latitude` and `longitude` fields, converted to `Marker`s.
- *
- * NOTE: This collects rows from all the collections the map is in, even if there's more than one.
- * At the time of writing this behavior won't be exercised since being included in multiple collections
- * is disabled in the UI.
- *
- * If we enable being in more than one collection again we should reconsider what behavior we actually want in that case.
+ * @returns Table rows in the same space as the map with `latitude` and `longitude` fields
  */
 export const useMarkers = (map: MapType): MapMarker[] => {
   const space = getSpace(map);
-  const collections = useQuery(
-    space,
-    Filter.schema(CollectionType, (collection) => collection.objects.some((obj) => obj?.id === map.id)),
-  );
 
   // TODO(burdon): Configure to select type.
-  // TODO(burdon): Why can obj be null?
-  const siblings = new Set(collections.flatMap((collection) => collection.objects.map((obj) => obj!.id)));
-  const tables = useQuery(
+  const schemas = useQuery(
     space,
-    Filter.schema(TableType, (table) => siblings.has(table.id) && table.schema && hasLocation(table.schema)),
-    undefined,
-    [JSON.stringify(Array.from(siblings))], // TODO(burdon): Hack.
+    Filter.schema(StoredSchema, (schema) => schema && hasLocation(schema)),
   );
 
-  const schemaIds: string[] = tables.flatMap((table) => (table.schema ? [table.schema.id] : []));
-
   const objects = useQuery(
-    // Short circuit if there are no `schemaIds` to find rows for:
-    schemaIds.length === 0 ? undefined : space,
-    // TODO(seagreen): performance issue since this checks every object.
-    //  Doing this with 'or's, e.g., Filter.or(...schemaIds.map((id) => Filter.typename(id)))
-    //  is forbidden since the number of 'or's will vary as `schemaIds` changes,
-    //  and the internals of `useQuery` require it to be fixed:
+    space,
     (obj) => {
+      const schemaNames = schemas.map((schema) => schema.id);
       const typename = getTypename(obj);
-      return typename ? schemaIds.includes(typename) : false;
+      return typename ? schemaNames.includes(typename) : false;
     },
     undefined,
-    [JSON.stringify(Array.from(schemaIds))], // TODO(burdon): Hack.
+    [schemas],
```

Comment:
@wittjosiah You mentioned the possibility of removing this `[schemas]` dep. At the moment if I remove it the map doesn't populate with icons, neither on initial load nor if you add new map data rows. Any idea what could be causing that?

A related question: is the expected behavior of `useQuery` to re-run when any data in the db that might affect the result of the query changes, even if the filter or dependency arguments haven't changed?

Reply:
In this case it's definitely needed because the query is just a callback. Where I was suggesting it might not be was when using `Filter` because iirc if the filter changes it would rerun the query

Reply:
Ah it makes sense now, thank you!